Resolve several compile warnings when comparing different int signedness

alsed opened the pull request:
advances #713 and eliminates several compilation warnings concerning comparison between integers of different signedness.

alsed committed:
var game_type converted to i32 in all instances

this resolve a compilation warning about comparison of expressions wih different signedness.

alsed committed:
explicit conversion to u32 of GetCurrentProcessId()/getpid() in Server_Client_Check function

this avoid a compilation warning about comparison of integer expressions of different signedness.

alsed committed:
local variable "c" converted to size_t

this avoid compilation warnings about comparison expressions of different signedness.

alsed committed:
var "m_style_id" explictly converted to int

- resolves a compilation warning about comparison expressions
of different signedness.
- setupstyle() uses getcurrentstylename() to retrieve name.
- correction in the message log.

alsed committed:
calculate_packed_xz() refactored in level_graph_inline.h

- resolved a comp warning about comparison of different signedness
- new function created: calculate_packed_xz(),
used by const CPosition& CLevelGraph::vertex_position() and bool CLevelGraph::inside()
- aditional minor resolves

alsed committed:
resolves several warning compilations when comparing differents signedness

- resolves a warning compilation in ConvertTextureFormat(),
modified to a for loop range, this avoid integers comparisions.

- resolves warnings when int comparing with unsigned macro R__NUM_SUN_CASCADES,
in render_phase_sun.cpp.

- resolves warnings when int comparing with unsigned macro R__NUM_PARALLEL_CONTEXTS,
in r2_loader.cpp

- var "b_count" converted to unsigned to resolve warnings within for loops,
in AnimationKeyCalculate.h

- resolves warnings when for loops compare signed vs unsigned, in r__sector.cpp

alsed committed:
Resolves several compile warnings when comparing different int signedness (2)

- advances #713 and eliminates all warnings concerning this operation:
- in R_dsgraph_structure::load() / r__dsgraph_build.cpp
- var 'vcnt' converted to u32 in R_Backend_DBG.cpp
- several for loops in light.cpp
- several for loops in in DetailManager_CACHE.cpp
- var 'filled_blocks' converted from int to i32, in SoundRender_Emitter.cpp
- in CSoundRender_Core::create() / SoundRender_Core.cpp,
explicit convertion of var 'sg_SourceType' to int
- for loop in CollectorPacked::CollectorPacked() / xrCDB_Collector.cpp
- several for loops in Frustum.cpp

alsed committed:
Merge branch 'OpenXRay:dev' into dev

alsed committed:
Merge branch 'OpenXRay:dev' into dev

Xottab-DUTY commented on src/Layers/xrRender/AnimationKeyCalculate.h:
```diff
@@ -178,7 +178,7 @@ IC void Dequantize(CKey& K, const CBlend& BD, const CMotion& M)
 }
 
-IC void MixInterlerp(CKey& Result, const CKey* R, const CBlend* const BA[MAX_BLENDED], int b_count)
+IC void MixInterlerp(CKey& Result, const CKey* R, const CBlend* const BA[MAX_BLENDED], unsigned int b_count)
```
Lets use u32 here

Xottab-DUTY commented on src/Layers/xrRender/R_Backend_DBG.cpp:
```diff
@@ -191,7 +191,7 @@ void CBackend::dbg_DrawEllipse(Fmatrix& T, u32 C)
-    const int vcnt = sizeof(gVertices) / (sizeof(float) * 3);
+    const u32 vcnt = sizeof(gVertices) / (sizeof(float) * 3);
     FVF::L verts[vcnt];
     for (int i = 0; i < vcnt; i++)
```
Might as well use `u32` instead of `int` in this cycle.

Xottab-DUTY commented on src/Layers/xrRender/light.cpp:
```diff
@@ -36,7 +36,7 @@ light::light() : SpatialBase(g_pGamePersistent->SpatialSpace)
     vis.visible = true;
     vis.pending = false;
-    for (int id = 0; id < R__NUM_CONTEXTS; ++id)
+    for (unsigned int id = 0; id < R__NUM_CONTEXTS; ++id)
```
Lets use u32 here

Xottab-DUTY commented on src/xrGame/xrServer.cpp:
```diff
@@ -866,7 +866,7 @@ void xrServer::Server_Client_Check(IClient* CL)
     };
 
-    if (CL->process_id == GetCurrentProcessId())
+    if (CL->process_id == static_cast<u32>(GetCurrentProcessId()))
```
On Windows `process_id` variable has `DWORD` type and `GetCurrentProcessId()` returns `DWORD` too.
On Linux this variable has type `pid_t` and GetCurrentProcessId() is a macro to `getpid` function that returns `pid_t`.

Why does it throw a warning here??

alsed replied:
As with all the changes in this commit, the warning is for comparing a signed int with an unsigned.

Edit:
Is for `pid_t` :
```
warning: comparison of integer expressions of different signedness: ‘u32’ {aka ‘unsigned int’} and ‘__pid_t’ {aka ‘int’} [-Wsign-compare]
  869 |     if (CL->process_id == GetCurrentProcessId())
      |         ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~``

Xottab-DUTY replied:
Try this. Will it fix the warning?
```suggestion
    if (CL->process_id == static_cast<xrpid_t>(GetCurrentProcessId()))
```

alsed replied:
The warning persist, xrpid_t is consider an int.
How about converting process_id to int for this comparison? unsigned to signed should be safer
`if (static_cast<int>(CL->process_id) == GetCurrentProcessId())`

Xottab-DUTY reviewed (changes requested):
LGTM!
I didn't checked xrAICore and xrSound yet.
The main thing to correct here is replace `unsigned int` with `u32`.

alsed committed:
Requested changes: "unsigned int" to "u32"

Also, added fixes for Images.cpp

alsed committed:
Merge branch 'OpenXRay:dev' into dev

Xottab-DUTY commented on src/xrAICore/Navigation/level_graph_inline.h:
```diff
@@ -80,19 +80,23 @@ ICF const Fvector& CLevelGraph::vertex_position(
+    const int packed_xz = iFloor((x - box_x) / cell_size + .5f) * m_row_length +
+		iFloor((z - box_z) / cell_size + .5f);
 
-    VERIFY(iFloor((source_position.z - box_z) / cell_size + .5f) < (int)m_row_length);
```
This VERIFY was removed. Needs to be returned.

alsed replied:
Ops! ...returned.

Xottab-DUTY commented on src/xrSound/SoundRender_Core.cpp:
```diff
@@ -115,7 +115,7 @@ CSound* CSoundRender_Core::create(pcstr fName, esound_type sound_type, int game_
 
     snd->g_type = game_type;
-    if (game_type == sg_SourceType)
+    if (game_type == static_cast<int>(sg_SourceType))
```
`game_type` is `u32`, so no need to cast `sg_SourceType` to `int`.
```suggestion
    if (game_type == sg_SourceType)
```

Xottab-DUTY commented on src/xrSound/SoundRender_Emitter.h:
```diff
@@ -85,7 +85,7 @@ class CSoundRender_Emitter final : public CSound_emitter
 
     size_t current_block{};
-    int filled_blocks{};
+    u32 filled_blocks{};
```
This was actually made as `int` on purpose in case of underflows.
It's better to leave it as is for now and examine the code that operates on this variable to make sure there are no underflows.
```suggestion
    int filled_blocks{};
```

alsed replied:
Alright, comment added because I have the memory of a goldfish

Xottab-DUTY replied:
No comment here, please.
Other person is working on xrSound currently and it might cause conflicts.

alsed replied:
Ok

alsed committed:
revision changes:

- all `unsigned int` changed to `u32`
- on SoundRender_Emitter.h - var `filled_blocks` reverted to `int` as requested.
- reverted changes to `level_graph_inline.h`
- proposal for `Server_Client_Check()` on `xrServer.cpp`

Xottab-DUTY commented on src/xrGame/xrServer.cpp:
```diff
@@ -866,7 +866,7 @@ void xrServer::Server_Client_Check(IClient* CL)
     };
 
-    if (CL->process_id == GetCurrentProcessId())
+    if (static_cast<int>(CL->process_id) == GetCurrentProcessId())
```
I'm only unsure about this, but whatever. It can be changed later, if needed.
